sled-agent/src/services.rs

```diff
@@ -495,65 +806,67 @@ impl ServiceManager {
     // more clearly.
     pub async fn load_services(&self) -> Result<(), Error> {
         let log = &self.inner.log;
-        let ledger_paths = self.all_service_ledgers().await;
-        info!(log, "Loading services from: {ledger_paths:?}");
-
         let mut existing_zones = self.inner.zones.lock().await;
         let Some(mut ledger) =
-            Ledger::<AllZoneRequests>::new(log, ledger_paths).await
+            self.load_ledgered_zones(&existing_zones).await?
         else {
-            info!(log, "Loading services - No services detected");
+            // Nothing found -- nothing to do.
+            info!(
+                log,
+                "Loading Omicron zones - \
+                no zones nor old-format services found"
+            );
             return Ok(());
         };
-        let services = ledger.data_mut();
+
+        let zones_config = ledger.data_mut();
+        info!(
+            log,
+            "Loaded Omicron zones";
+            "zones_config" => ?zones_config
+        );
+        let omicron_zones_config =
+            zones_config.clone().to_omicron_zones_config();
 
         // Initialize internal DNS only first: we need it to look up the
         // boundary switch addresses. This dependency is implicit: when we call
-        // `ensure_all_services` below, we eventually land in
+        // `ensure_all_omicron_zones` below, we eventually land in
         // `opte_ports_needed()`, which for some service types (including Ntp
         // but _not_ including InternalDns), we perform internal DNS lookups.
```

Comment:
I see what you mean.  That this works as-is suggests that the comment is wrong.  But in debugging the symptoms that led to oxidecomputer/meta#324, I found that `opte_ports_needed()` does do DNS lookups when processing the boundary NTP zone (and not the internal DNS zone).  It may be that this only works because the internal DNS zones appear first (which in turn is because the sled plan happens to put them first and we preserve this property in RSS -- discussed elsewhere in this PR).  This is all kind of a mess related to https://github.com/oxidecomputer/omicron/pull/4466#discussion_r1409586932 and I think it's worth cleaning up...but I think this PR preserves the existing behavior.